src/main/java/seedu/taassist/logic/commands/DeletecCommand.java

```diff
@@ -41,6 +45,18 @@ public CommandResult execute(Model model) throws CommandException {
                     model.getModuleClassList()));
         }
 
+        List<Student> students = model.getFilteredStudentList();
+        for (Student student : students) {
+            if (!Collections.disjoint(student.getModuleClasses(), moduleClasses)) {
+                Set<ModuleClass> assignedClasses = new HashSet<>();
+                assignedClasses.addAll(student.getModuleClasses());
+                assignedClasses.removeAll(moduleClasses);
```

Comment:
I believe this doesn't achieve the intended behavior? 

In this case, comparing equality should use the `ModuleClass::isSame` method. Doing `removeAll` in this scenario will check by `ModuleClass::equals` and remove it based on that. Perhaps this is also a design issue that can be looked into again.

Reply:
After further thought. This design is fine for now, since `Student`-s always store moduleClasses with empty `List<Sessions>` in practice. However, this could end up being confusing in the long run. We'll probably need to introduce some interface to be returned instead of a `Set<ModuleClass>`. Will KIV in a future issue. This comment is resolved for now.